src/lib/utilities.js

```diff
@@ -382,6 +383,12 @@ module.exports = {
 			return new Error(_.isArray(err) ? err.join('\n') : err);
 		}
 		return err;
+	},
+
+	parsePropertyFile(propPath) {
+		const savedProp = fs.readFileSync(propPath, 'utf8');
```

Comment:
I'd rather use `await fs.readFile(...)` instead of `readFileSync` the reason is: as far I know `readFileSync` stops the whole eventLoop until it completes the action.

Here's an example of how to use the `readFile` 👉  https://github.com/particle-iot/particle-cli/blob/master/src/cmd/binary.js#L61

Reply:
That way my initial idea but there are other places in the file which are using readFileSync so I kept it consistent with them other than a readFile which waits for the promise to be resolved.

Reply:
We have lots of fs sync usage in the CLI. It's best to clean it up when we can but async is viral, so changing one function to async might mean a bunch of changes in other parts of the code.

The consequences are minor in the CLI since it's a single user program so I think that's a chore for another day.